src/effects/something_rotating.c

```diff
@@ -66,7 +126,111 @@ EffectInstance* something_rotating_main(
 void something_rotating_init(EffectInstance* effect) {
 }
 
-INCLUDE_ASM(s32, "effects/something_rotating", something_rotating_update);
+void something_rotating_update(EffectInstance* effect) {
+    SomethingRotatingFXData* part = effect->data.somethingRotating;
+    f32 unk_04 = part->unk_04;
+    f32 unk_08 = part->unk_08;
+    f32 unk_0C = part->unk_0C;
+    s32 unk_14;
+    s32 unk_14_2;
+    f32 factor;
+    f32 angle1;
+    s32 i;
+
+    if (effect->flags & 0x10) {
+        effect->flags &= ~0x10;
+        part->unk_10 = 32;
+    }
+
+    if (part->unk_10 < 1000) {
+        part->unk_10--;
+    }
+
+    part->unk_14++;
+
+    if (part->unk_10 < 0) {
+        shim_remove_effect(effect);
+        return;
+    }
+
+    unk_14 = part->unk_14;
+    part->unk_20 = 90.0f;
+    angle1 = 90.0f;
+
+    part++;
+    for (i = 1; i < effect->numParts; i++, part++) {
+        f32 angle2 = unk_14 * 4.0f + (f32) i * 51.43;
+        f32 temp_f24 = shim_sin_deg(angle2);
+        f32 temp_f22 = part->unk_2C;
+
+        part->unk_04 = unk_04 + temp_f22 * shim_sin_deg(angle1) * temp_f24;
+        part->unk_08 = unk_08 + temp_f22 * shim_cos_deg(angle2);
+        part->unk_0C = unk_0C + temp_f22 * shim_cos_deg(angle1) * temp_f24;
+
+        switch (part->unk_29 - 1) {
+            case 0:
+                part->unk_14 = 0;
+                part->unk_29++;
+                // fallthrough
+            case 1:
+                unk_14_2 = part->unk_14;
+
+                if (unk_14_2 < ARRAY_COUNT(D_E0116CA4)) {
+                    part->unk_20 = D_E0116CA4[unk_14_2];
+                } else {
+                    part->unk_20 = D_E0116CA4[ARRAY_COUNT(D_E0116CA4) - 1];
+                }
+                if (unk_14_2 < ARRAY_COUNT(D_E0116CF8)) {
+                    part->unk_24 = D_E0116CF8[unk_14_2];
+                } else {
+                    part->unk_24 = D_E0116CF8[ARRAY_COUNT(D_E0116CF8) - 1];
+                }
+                if (unk_14_2 < ARRAY_COUNT(D_E0116D08)) {
+                    part->unk_25 = D_E0116D08[unk_14_2];
+                } else {
+                    part->unk_25 = D_E0116D08[ARRAY_COUNT(D_E0116D08) - 1];
+                }
+
+                if (unk_14_2 < ARRAY_COUNT(D_E0116D3C)) {
+                    factor = D_E0116D3C[unk_14_2];
+                } else {
+                    factor = D_E0116D3C[ARRAY_COUNT(D_E0116D3C) - 1];
+                }
+
+                factor *= 0.0039215686; // 1 / 255
```

Comment:
does 1.0 / 255 work? you might need to fudge the numbers a tiny bit, but that seems better than the strange constant

Reply:
It doesn't, unfortunately. 1.0 / 255 is more precise: 0.00392156862745098

Reply:
Ahhh okh